### PR TITLE
Added support for UniqueValue renderer in ArcGIS feature format

### PR DESCRIPTION
**Pull request by mdastous-bentley**
Implemented ArcGIS's [UniqueValue renderer](https://developers.arcgis.com/web-map-specification/objects/uniqueValueRenderer/) to allow features to be symbolized based on attribute values.

So instead having a simple renderer where all features are symbolized the same way, we can have a dynamic symbolization:
image

This apply to line, polygon and symbol marker symbolization.

### CODE REVIEW

**Review by pmconne — COMMENTED** [review]
Run `rush change`, commit, and push.
This will also trigger the builds that didn't start when you marked ready for review.

File: core/frontend/src/tile/map/ImageryProviders/ArcGisAttributeDrivenSymbology.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
```
[the remark this answers is not in the document]
Comment (pmconne) [reply core/frontend/src/tile/map/ImageryProviders/ArcGisAttributeDrivenSymbology.ts:5]:
```suggestion
/** @packageDocumentation
 * @module Tiles
 */

```

File: extensions/map-layers-formats/src/ArcGisFeature/ArcGisCanvasRenderer.ts [hunk 2]
```diff
@@ -5,12 +5,17 @@
...
-import { ArcGisGeometryBaseRenderer } from "@itwin/core-frontend";
+import { ArcGisAttributeDrivenSymbology, ArcGisGeometryBaseRenderer } from "@itwin/core-frontend";
 
+/** @internal */
 export class ArcGisCanvasRenderer extends ArcGisGeometryBaseRenderer {
   private _context: CanvasRenderingContext2D;
   private _symbol: ArcGisSymbologyRenderer;
 
+  public override get attributeSymbology() {
+    return this._symbol.attributeDriven ? (this._symbol as unknown) as ArcGisAttributeDrivenSymbology : undefined;
```
Comment (pmconne) [inline extensions/map-layers-formats/src/ArcGisFeature/ArcGisCanvasRenderer.ts:16]: Motivation for this cast?

**Review by pmconne — APPROVED** [review]

File: extensions/map-layers-formats/src/ArcGisFeature/ArcGisCanvasRenderer.ts (hunk 2 above)
Comment (mdastous-bentley) [reply extensions/map-layers-formats/src/ArcGisFeature/ArcGisCanvasRenderer.ts:16]: If the `attributeDriven` returns true, I know I have a [ArcGisUniqueValueSymbologyRenderer](https://github.com/iTwin/itwinjs-core/blob/ea715eb692b2d8e2bdb330ad7685956230a19d55/extensions/map-layers-formats/src/ArcGisFeature/ArcGisSymbologyRenderer.ts#L136), but ALSO an `ArcGisAttributeDrivenSymbology`, which **matters the most to me** :
`export class ArcGisUniqueValueSymbologyRenderer extends ArcGisSimpleSymbologyRenderer implements ArcGisAttributeDrivenSymbology`

Typescript would let me cast it as a `ArcGisSimpleSymbologyRenderer` (the base abstract class), but not the `ArcGisAttributeDrivenSymbology` it implements.  So I have to make a 'reinterpret' cast.

I feel like typescript is limiting polymorphism here, but maybe I just don't understand how it's meant to be used?

**Review by aruniverse — APPROVED** [review]
should add @eringram as a codeowner to the map-layers-* pkgs as well

File: extensions/map-layers-formats/src/ArcGisFeature/ArcGisCanvasRenderer.ts (hunk 2 above)
Comment (pmconne) [reply extensions/map-layers-formats/src/ArcGisFeature/ArcGisCanvasRenderer.ts:16]: Did you try something like:
```
export abstract class ArcGisSymbologyRenderer {
  public abstract isAttributeDriven(): this is ArcGisAttributeDrivenSymbology;
 // ...
}
```
Then:
```
export class ArcGisCanvasRenderer extends ArcGisGeometryBaseRenderer {
  public override get attributeSymbology(): ArcGisAttributeDrivenSymbology | undefined {
    return this._symbol.isAttributeDriven() ? this._symbol : undefined;
  }
  // ...
}
```

File: core/frontend/src/tile/map/ImageryProviders/ArcGisAttributeDrivenSymbology.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (mdastous-bentley) [reply core/frontend/src/tile/map/ImageryProviders/ArcGisAttributeDrivenSymbology.ts:8]: done

File: extensions/map-layers-formats/src/ArcGisFeature/ArcGisCanvasRenderer.ts (hunk 2 above)
Comment (mdastous-bentley) [reply extensions/map-layers-formats/src/ArcGisFeature/ArcGisCanvasRenderer.ts]: done, tx for the suggestion!

**Comment by pmconne** [comment]
> should add @eringram as a codeowner to the map-layers-* pkgs as well

Done.

**Pull request merged**